perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/widgets/items/validators/PerunFormItemValidatorImpl.java

```diff
@@ -41,7 +41,21 @@ public String getErrorMsgOrDefault(T item) {
 		}
 
 		if (errorText == null || errorText.isEmpty()) {
-			return translation.incorrectFormat();
+			switch (item.getItemData().getFormItem().getType()) {
```

Comment:
I'm not sure we need this for all of the form item types. This message is reserved for the cases, when regex validation of the form item fails. If you expect non-empty value (selected at least one option), it should be done through required flag of the form item. Emptiness is checked before the regex and should display correct messages for the form items (about selecting the option).

I found only one case (but maybe there are more) when we resort to this error message instead and its checkbox item with the single checkbox option. In such a case we prefer to display this customized error message (like we expect you to say "you need to approve the AUP" or whatever).

Reply:
You are right, I have not noticed that. So In the amended commit, I will add just an alternative translation for that edge-case scenario :).